### PR TITLE
Setup GitHub Actions for Unittests

### LINKED ISSUE #754: [FeatureRequest] Python 3.9 CI pipeline
Create a CI pipeline for Python 3.9.

Officially launched: https://www.python.org/downloads/release/python-390/
More Information: https://docs.python.org/release/3.9.0/whatsnew/3.9.html

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
This PR adds the Unittest CI pipeline for Python 3.9 (and other Python versions as well). This would help in moving the Unittest pipeline out of the Azure Pipelines workflow, speeding up runs, and keeping the results within the GitHub view.

Fixes #754

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        env:
+          VALIDATE_PYTHON: true
+          VALIDATE_PYTHON_FLAKE8: true
+          # VALIDATE_PYTHON_PYLINT: false # disable pylint, as we have not configure it
+          # VALIDATE_PYTHON_BLACK: false # same as above
+          PYTHON_FLAKE8_CONFIG_FILE: tox.ini
+          FILTER_REGEX_INCLUDE: azure_functions_worker/.*
+          FILTER_REGEX_EXCLUDE: tests/.*
+          DEFAULT_BRANCH: dev
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Hazhzeng) [inline .github/workflows/linter.yml:54]: May need to set the GITHUB_TOKEN secrets in This Repo -> Settings -> Secrets.

File: .github/workflows/unittesting.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: CI Unit tests
+
+on:
+  push:
+    branches: [ dev, master, main ]
```
Comment (Hazhzeng) [inline .github/workflows/unittesting.yml:8]: Could you remind me what's the `main` branch does? Are we migrating master to main?

**Review by Hazhzeng — APPROVED** [review]

File: .github/workflows/linter.yml (hunk 1 above)
Comment (vrdmr) [reply .github/workflows/linter.yml:54]: We dont need to pass as the workflow is running within GH.

From the documentation on [How to use the super linter](https://github.com/marketplace/actions/super-linter#how-to-use)?
> NOTE: If you pass the Environment variable GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} in your workflow, then the GitHub Super-Linter will mark the status of each individual linter run in the Checks section of a pull request. Without this you will only see the overall status of the full run. There is no need to set the GitHub Secret as it is automatically set by GitHub, it only needs to be passed to the action.

File: .github/workflows/unittesting.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    - name: Set up Dotnet 3.1.x
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: '3.1.x'
+    - name: Install dependencies and the worker
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple -U -e .[dev]
+        python -m pip install --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple -U azure-functions --pre
+        python setup.py webhost
```
Comment (Hazhzeng) [inline .github/workflows/unittesting.yml:34]: also need to setup `python setuup.py build` to compile protobuf.

File: .github/workflows/unittesting.yml (hunk 2 above)
Comment (vrdmr) [reply .github/workflows/unittesting.yml:8]: We would be migrating soon from master to main and deprecate the master branch - ref: https://github.com/github/renaming

**Pull request merged**